### PR TITLE
Extract RdContext into a base class to make value propert virtual

### PR DESCRIPTION
**Pull request by nerzhulart**

### CODE REVIEW

File: rd-kt/rd-gen/src/main/kotlin/com/jetbrains/rd/generator/nova/Types.kt
Comment (ForNeVeR) [inline rd-kt/rd-gen/src/main/kotlin/com/jetbrains/rd/generator/nova/Types.kt:450]: Could we please have an old function (which will delegate to `threadLocalContext`) as `@Deprecated`? Breaking changes are always very painful, and it's hard to apply them across all the codebase.

**Review by ForNeVeR — CHANGES_REQUESTED** [review]
In principle, LGTM. But let's wait for a review from @Iliya-usov.

**Review by ForNeVeR — APPROVED** [review]

**Review by Iliya-usov — APPROVED** [review]

**Pull request merged**
